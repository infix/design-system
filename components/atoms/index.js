import Button from './Button'
import Input from './Input'
import Heading from './Heading'
import Text from './Text'
import IconButton from './IconButton'
import { RadioGroup, RadioButton } from './Radio'
import { Select } from './Select'

import { TwitterShareButton, FacebookShareButton } from './ShareButton'

const atoms = {
  Text,
  Heading,
  Button,
  IconButton,
  Input,
  RadioGroup,
  RadioButton,
  Select,
  TwitterShareButton,
  FacebookShareButton
}

export default atoms
