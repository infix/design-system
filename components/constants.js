/* global module */
const defaultWeights = [
  400,
  700
]

const defaultBreakpoints = [
  40,
  52,
  64
]

const defaultSpace = [
  0,
  8,
  16,
  32,
  64
]

const defaultFontSizes = [
  12,
  14,
  16,
  20,
  24,
  32,
  48,
  64,
  72
]

module.exports = {
  defaultBreakpoints,
  defaultSpace,
  defaultFontSizes,
  defaultWeights
}
