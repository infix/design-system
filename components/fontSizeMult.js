/* global require */
import {
  style,
  px
} from 'styled-system'

const {
  defaultFontSizes
} = require('./constants')

export const fontSizeMult = (propName, k) => {
  /* eslint-disable no-console */
  console.error('usage of fontSizeMult will be deprecated')
  /* eslint-enable no-console */

  return style({
    prop: 'fontSize',
    cssProperty: propName,
    key: 'fontSizes',
    transformValue: n => px(n * k),
    scale: defaultFontSizes
  })
}

export default fontSizeMult
