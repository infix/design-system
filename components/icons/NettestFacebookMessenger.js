import React from 'react'
import Icon from 'react-icon-base'
const NettestFacebookMessenger = props => (
  <Icon viewBox="0 0 30 30" {...props}>
    <g>
  <defs>
    
  </defs>
  <title>Facebook Messenger</title>
  <g>
    <path d="M15.5,17.305,12.939,14.56l-4.99,2.745,5.49-5.864,2.62,2.745,4.928-2.745Zm-1-12.6C8.953,4.7,4.455,8.9,4.455,14.061A9.115,9.115,0,0,0,8.2,21.348v3.568l3.419-1.89a10.641,10.641,0,0,0,2.882.393c5.546,0,10.044-4.193,10.044-9.358S20.045,4.7,14.5,4.7Z"/>
  </g>
</g>
  </Icon>
)
export default NettestFacebookMessenger
