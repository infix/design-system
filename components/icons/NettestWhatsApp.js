import React from 'react'
import Icon from 'react-icon-base'
const NettestWhatsApp = props => (
  <Icon viewBox="0 0 30 30" {...props}>
    <g>
  <defs>
    
  </defs>
  <title>WhatsApp</title>
  <g>
    <path d="M14.834,3.23a10.547,10.547,0,0,1,4.215.865A10.736,10.736,0,0,1,24.835,9.88a10.7,10.7,0,0,1,0,8.43A10.733,10.733,0,0,1,19.049,24.1a10.564,10.564,0,0,1-4.215.865,10.69,10.69,0,0,1-5.26-1.355L3.565,25.537,5.525,19.7A10.8,10.8,0,0,1,4.834,9.88,10.739,10.739,0,0,1,10.619,4.1,10.547,10.547,0,0,1,14.834,3.23Zm0,19.915a8.813,8.813,0,0,0,3.509-.721,8.913,8.913,0,0,0,4.82-4.82,8.9,8.9,0,0,0,0-7.018,8.913,8.913,0,0,0-4.82-4.82,8.911,8.911,0,0,0-7.018,0,8.913,8.913,0,0,0-4.82,4.82A8.964,8.964,0,0,0,7.514,19.4L6.375,22.756l3.488-1.11A8.857,8.857,0,0,0,14.834,23.145Zm2.925-7.637a6.5,6.5,0,0,1,1.405.634,6.208,6.208,0,0,1,1.29.763.618.618,0,0,1,.029.216,3.052,3.052,0,0,1-.245,1.1,1.938,1.938,0,0,1-1.023.944,3.409,3.409,0,0,1-1.47.381,7.792,7.792,0,0,1-2.738-.893,8.233,8.233,0,0,1-2.45-1.7,19.069,19.069,0,0,1-2.132-2.666,5.03,5.03,0,0,1-1.023-2.8v-.115A3.181,3.181,0,0,1,10.468,9.1a1.085,1.085,0,0,1,.749-.317,2.194,2.194,0,0,1,.26.022,2.39,2.39,0,0,0,.273.021.593.593,0,0,1,.382.094,1.037,1.037,0,0,1,.224.4q.114.288.475,1.268a8.183,8.183,0,0,1,.36,1.081,1.459,1.459,0,0,1-.5.829q-.5.525-.5.67a.422.422,0,0,0,.072.216,6.492,6.492,0,0,0,1.47,1.974A9.2,9.2,0,0,0,15.915,16.8a.629.629,0,0,0,.317.1q.216,0,.778-.7T17.759,15.508Z"/>
  </g>
</g>
  </Icon>
)
export default NettestWhatsApp
