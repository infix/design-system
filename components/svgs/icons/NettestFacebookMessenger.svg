<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="30" height="30" viewBox="0 0 30 30">
  <defs>
    <clipPath id="clip-path">
      <rect width="30" height="30" style="fill: none"/>
    </clipPath>
  </defs>
  <title>Facebook Messenger</title>
  <g style="clip-path: url(#clip-path)">
    <path d="M15.5,17.305,12.939,14.56l-4.99,2.745,5.49-5.864,2.62,2.745,4.928-2.745Zm-1-12.6C8.953,4.7,4.455,8.9,4.455,14.061A9.115,9.115,0,0,0,8.2,21.348v3.568l3.419-1.89a10.641,10.641,0,0,0,2.882.393c5.546,0,10.044-4.193,10.044-9.358S20.045,4.7,14.5,4.7Z"/>
  </g>
</svg>
