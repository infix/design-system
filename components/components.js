import atoms from './atoms'
import molecules from './molecules'
import organisms from './organisms'

import BarChart from './charts/BarChart'
import PieChart from './charts/PieChart'

/* OONI produced atoms */

export const Button = atoms.Button
export const Input = atoms.Input
export const RadioButton = atoms.RadioButton
export const RadioGroup = atoms.RadioGroup
export const TwitterShareButton = atoms.TwitterShareButton
export const FacebookShareButton = atoms.FacebookShareButton
export const Heading = atoms.Heading
export const Text = atoms.Text
export const Select = atoms.Select

/* OONI produced molecules */
export const InputWithIconButton = molecules.InputWithIconButton
export const LogoOONIRun = molecules.LogoOONIRun
export const Card = molecules.Card

/* OONI produced organisms */
export const Hero = organisms.Hero
export const HeroLead = organisms.HeroLead
export const Modal = organisms.Modal

/* Here we just export rebass until we implement all of these */
import {
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Checkbox,
  Circle,
  Close,
  Code,
  Column,
  Container,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Label,
  Lead,
  Link,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Radio,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Slider,
  Small,
  Star,
  Sticky,
  Subhead,
  Switch,
  TabItem,
  Tabs,
  Textarea,
  Toolbar,
  Tooltip,
  Truncate
} from 'rebass'

export { default as BarChart } from './charts/BarChart'
export { default as PieChart } from './charts/PieChart'

const components = {
  Button,
  Heading,
  Text,
  Input,
  Select,
  Card,
  RadioButton,
  RadioGroup,
  InputWithIconButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart,
  Absolute,
  Arrow,
  Avatar,
  BackgroundImage,
  Badge,
  Banner,
  BlockLink,
  Blockquote,
  Border,
  ButtonCircle,
  ButtonOutline,
  ButtonTransparent,
  Carousel,
  CarouselSlide,
  Checkbox,
  Circle,
  Close,
  Code,
  Column,
  Container,
  Divider,
  Donut,
  DotButton,
  Drawer,
  Embed,
  Fixed,
  Group,
  Image,
  Label,
  Lead,
  Link,
  Measure,
  Media,
  Message,
  NavLink,
  Overlay,
  Panel,
  PanelFooter,
  PanelHeader,
  Pre,
  Progress,
  Radio,
  Relative,
  Row,
  Samp,
  ScrollCarousel,
  Slider,
  Small,
  Star,
  Sticky,
  Subhead,
  Switch,
  TabItem,
  Tabs,
  Textarea,
  Toolbar,
  Tooltip,
  Truncate
}

export default components
