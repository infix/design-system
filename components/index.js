import {
  Provider,
  Flex,
  Box
} from 'rebass'

import colors from './theme/colors'
import theme from './theme/rebassTheme'

export { default as colors } from './theme/colors'
export { default as theme } from './theme/rebassTheme'

import components from './components'

// We export these from rebass directly
export {
  Provider,
  Flex,
  Box
} from 'rebass'

export const {
  Button,
  ButtonOutline,
  ButtonCircle,
  ButtonTransparent,
  Link,
  NavLink,
  BlockLink,
  Heading,
  Subhead,
  Text,
  Small,
  Lead,
  Pre,
  Code,
  Samp,
  Blockquote,
  Measure,
  Truncate,
  Label,
  Input,
  Select,
  Textarea,
  Checkbox,
  Radio,
  Slider,
  Image,
  Avatar,
  BackgroundImage,
  Container,
  Divider,
  Border,
  Media,
  Card,
  Banner,
  Panel,
  PanelHeader,
  PanelFooter,
  Progress,
  Message,
  Group,
  Toolbar,
  Badge,
  Circle,
  Tabs,
  TabItem,
  DotButton,
  Close,
  Relative,
  Absolute,
  Fixed,
  Sticky,
  Drawer,
  Overlay,
  Carousel,
  ScrollCarousel,
  CarouselSlide,
  Tooltip,
  Switch,
  Arrow,
  Star,
  Embed,
  Donut,
  Row,
  Column,
  RadioGroup,
  RadioButton,
  HeroLead,
  Hero,
  LogoOONIRun,
  InputWithIconButton,
  TwitterShareButton,
  FacebookShareButton,
  Modal,
  BarChart,
  PieChart
} = components

export const OONIComponents = Object.assign({}, components, { Provider, Box, Flex, colors, theme })

export default OONIComponents
