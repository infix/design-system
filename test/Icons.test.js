/* global describe, test, expect */
import React from 'react'
import { renderJSON } from './index'
import { CategoryCodePOLR, NettestGroupWebsites } from '../components/icons'

describe('Icons', () => {
  test('renders Category Code Icon', () => {
    const json = renderJSON(
      <CategoryCodePOLR />
    )
    expect(json).toMatchSnapshot()
  })

  test('renders icon for Test Group', () => {
    const json = renderJSON(
      <NettestGroupWebsites />
    )
    expect(json).toMatchSnapshot()
  })
})
