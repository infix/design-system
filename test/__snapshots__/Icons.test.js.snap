// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons renders Category Code Icon 1`] = `
<CategoryCodePOLR>
  <IconBase
    viewBox="0 0 16 16"
  >
    <svg
      fill="currentColor"
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "color": undefined,
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 16 16"
      width="1em"
    >
      <g>
        <title>
          CategoryCodePOLR
        </title>
        <path
          d="M13.329,7.056a.868.868,0,0,0,.647-1.041.869.869,0,0,0-1.039-.653L12.35,2.821a.88.88,0,0,0-1.038-.652A9.559,9.559,0,0,1,6,6.073l-3.16.73a1.037,1.037,0,0,0-.672.483,1.049,1.049,0,0,0-.137.82l.294,1.27a1.049,1.049,0,0,0,.482.676,1.037,1.037,0,0,0,.816.139s.42-.083.977-.2l1.49,3.213a.837.837,0,0,0,1,.625l.695-.16a.838.838,0,0,0,.625-1l-1.3-3.232a1.27,1.27,0,0,0,.233-.078,10.014,10.014,0,0,1,5.928,1.279A.881.881,0,0,0,13.916,9.6ZM7.44,8.466,7.027,6.679a10.492,10.492,0,0,0,4.532-3.438l1.459,6.313A10.523,10.523,0,0,0,7.44,8.466Z"
        />
      </g>
    </svg>
  </IconBase>
</CategoryCodePOLR>
`;

exports[`Icons renders icon for Test Group 1`] = `
<NettestGroupWebsites>
  <IconBase
    viewBox="0 0 30 30"
  >
    <svg
      fill="currentColor"
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "color": undefined,
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 30 30"
      width="1em"
    >
      <g>
        <defs />
        <title>
          Websites
        </title>
        <g>
          <path
            d="M19.347,16.686a17.92,17.92,0,0,0,.157-2.3,22.427,22.427,0,0,0-.157-2.3h3.895a9.839,9.839,0,0,1,.295,2.3,9.841,9.841,0,0,1-.295,2.3Zm-2.026,6.393a18.452,18.452,0,0,0,1.593-4.092H22.3A9.221,9.221,0,0,1,17.321,23.079Zm-.3-6.393h-5.39a17.744,17.744,0,0,1-.177-2.3,17.474,17.474,0,0,1,.177-2.3h5.39a17.646,17.646,0,0,1,.177,2.3A17.922,17.922,0,0,1,17.026,16.686Zm-2.695,6.865a15.783,15.783,0,0,1-2.2-4.564h4.386A15.5,15.5,0,0,1,14.331,23.551ZM9.767,9.781h-3.4A9.225,9.225,0,0,1,11.341,5.69,17.424,17.424,0,0,0,9.767,9.781Zm-3.4,9.206h3.4a18.452,18.452,0,0,0,1.593,4.092A9.2,9.2,0,0,1,6.364,18.987Zm-.944-2.3a9.841,9.841,0,0,1-.295-2.3,9.839,9.839,0,0,1,.295-2.3H9.315a17.738,17.738,0,0,0-.158,2.3,22.148,22.148,0,0,0,.158,2.3ZM14.331,5.218a15.815,15.815,0,0,1,2.2,4.563H12.128A16.03,16.03,0,0,1,14.331,5.218ZM22.3,9.781h-3.4A18.4,18.4,0,0,0,17.3,5.69,9.2,9.2,0,0,1,22.3,9.781Zm-7.986-6.9A11.508,11.508,0,1,0,25.838,14.384,11.493,11.493,0,0,0,14.311,2.877Z"
          />
        </g>
      </g>
    </svg>
  </IconBase>
</NettestGroupWebsites>
`;
